templates/CRM/common/paymentBlock.tpl

```diff
@@ -95,7 +95,10 @@
 
       var payment_instrument_id = $('#payment_instrument_id').val();
 
-      var dataUrl = "{crmURL p='civicrm/payment/form' h=0 q="formName=`$form.formName`&currency=`$currency`&`$urlPathVar``$isBackOfficePathVar``$profilePathVar``$contributionPageID``$preProfileID`processor_id="}" + type;
+      var currency = '{$currency}';
+      currency = currency == '' ? $('#currency').val() : currency;
```

Comment:
This is the only bit I have concerns with - why would the currency not be set when the form loads?

Reply:
@mattwire actually there are two cases where the currency variable is not set in backend, these are:
 
1. When we open the standalone CC form for submitting new live contribution, where we are setting a NULL currency value in [here](https://github.com/civicrm/civicrm-core/blob/master/CRM/Contribute/Form/Contribution.php#L429) Although the default currency set in the system can be used in here instead if NULL. So I am not sure if we need to do that.

2. When we are changing the  currency beside the amount field https://github.com/civicrm/civicrm-core/pull/18517/files#diff-726c3dcc8c2364a0b9b8f0c0e11bcd34R122 which call the js function, and in this case the code should consider what currency value has been chosen in b/o form and when in reloads the snipper 

This is not a issue on edit CC form where currency variable is set.

Reply:
@monishdeb can we pass currency though the url for case 2?

Reply:
@eileenmcnaughton yes I am passing the currency value for the 2nd case, if you look  at L122 below,  I am calling the buildPaymentBlock() fn on currency change and eventually at L99 we are considering the currency value and passing through the url  at L144